client/clientui.py

```diff
@@ -270,46 +272,77 @@ def update_compass(self, compass_update):
             return [[x - 3, y + 4, x + 3, y - 4],
                     [x - 3, y + 3, x + 3, y - 3],
-                    [x - 3, y + 2, x + 3, y - 2]]
+                    [x - 3, y + 1, x + 3, y - 1]]
         elif direction == "nw" or direction == "se":
             return [[x - 3, y - 4, x + 3, y + 4],
                     [x - 3, y - 3, x + 3, y + 3],
-                    [x - 3, y - 2, x + 3, y + 2]]
+                    [x - 3, y - 1, x + 3, y + 1]]
 
     def update_map(self, map_elements):
         self.map_area.delete("all")
         for position in map_elements:
             size = int(position[2])
-            x = int(position[0]) + 60
-            y = int(position[1]) + 60 + size
+            x = int(position[0]) + self.MAP_OFFSET
+            y = int(position[1]) + self.MAP_OFFSET + size
             self.map_area.create_rectangle(x, y, x + size, y - size, fill=position[3])
 
     def create_map_area(self, side_bar):
         self.map_area = tk.Canvas(side_bar, name="map", width=120, height=120, bg='black')
         self.map_area.pack(side='bottom')
 
+    def create_macro_area(self, side_bar):
+
+        macros = tk.Frame(side_bar, name="macros", width=120, height=120, bg='black')
+
+        tk.Button(macros, text='I', command=lambda: self.send_command("fe1")).grid(row=0, column=0, sticky='WENS')
+        tk.Button(macros, text='II', command=lambda: self.send_command("fe2")).grid(row=0, column=1, sticky='WENS')
+        tk.Button(macros, text='III', command=lambda: self.send_command("fe3")).grid(row=0, column=2, sticky='WENS')
+        tk.Button(macros, text='IV', command=lambda: self.send_command("fe4")).grid(row=0, column=3, sticky='WENS')
+        tk.Button(macros, text='V', command=lambda: self.send_command("fe5")).grid(row=0, column=4, sticky='WENS')
+
+        tk.Button(macros, text='VI', command=lambda: self.send_command("fe6")).grid(row=1, column=0, sticky='WENS')
+        tk.Button(macros, text='VII', command=lambda: self.send_command("fe7")).grid(row=1, column=1, sticky='WENS')
+        tk.Button(macros, text='VIII', command=lambda: self.send_command("fe8")).grid(row=1, column=2, sticky='WENS')
+        tk.Button(macros, text='IX', command=lambda: self.send_command("fe9")).grid(row=1, column=3, sticky='WENS')
+        tk.Button(macros, text='X', command=lambda: self.send_command("fe10")).grid(row=1, column=4, sticky='WENS')
+
+        tk.Button(macros, text='XI', command=lambda: self.send_command("fe11")).grid(row=2, column=0, sticky='WENS')
+        tk.Button(macros, text='XII', command=lambda: self.send_command("fe12")).grid(row=2, column=1, sticky='WENS')
+        tk.Button(macros, text='XIII', command=lambda: self.send_command("fe13")).grid(row=2, column=2, sticky='WENS')
+        tk.Button(macros, text='XIV', command=lambda: self.send_command("fe14")).grid(row=2, column=3, sticky='WENS')
+        tk.Button(macros, text='XV', command=lambda: self.send_command("fe15")).grid(row=2, column=4, sticky='WENS')
+
```

Comment:
All the clients I see have 15 macro buttons and the commands `fe1` through `fe15` work. Going above that to `fe16` through `fe20` failed in my testing.

    "fe20" matches no commands.